libvast/vast/index_config.hpp

```diff
@@ -0,0 +1,53 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2022 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "vast/type.hpp"
+
+#include <string>
+#include <vector>
+
+namespace vast {
+
+/// The configuration for `vast.index`.
```

Comment:
What is `vast.index`? Do you mean `vast::system::index`? Or is it the configuration under the key `vast.index`?

Reply:
It's the new configuration for anything index-related in VAST, per our internal design doc. 

It shouldn't be in namespace `system` because there are top-level components (e.g., sketch/partition) that use it.

Reply:
Should the comment then say "the configuration _under_ vast.index"?

Reply:
I think you can argue both ways. It's the data structure "for" the configuration key `vast.index`. I don't see the hierarchy aspect that "under" implies.

Reply:
Or "for the configuration key vast.index"? That'd at least resolve the ambiguity.

Reply:
I disarmed this discussion by removing the `vast.index` part entirely.

Reply:
That also works for me!